### DISCUSSION TITLE
DGEMM error when turning on adaptivity

### DISCUSSION THREAD

**Discussion by souzanha** [idaholab/moose#17432] [Q&A General] [answered]
Hi! 
I have a code that converges and runs quite well using the phase-field module. However, when I turn on adaptivity I get an error at the 3rd timestep that I'm not sure how to deal with:

** On entry to DGEMM parameter number  8 had an illegal value

Conda and moose is updated.
This is my block:

```[Executioner]
  type = Transient
  scheme = 'BDF2'

  solve_type = 'NEWTON'

  petsc_options_iname = '-pc_type -sub_pc_type'
  petsc_options_value = 'asm       ilu'

  l_tol = 1.0e-4
  nl_rel_tol = 1.0e-10
  nl_abs_tol = 1.0e-10

  dt = 1e-3
  end_time = 1

  [./Adaptivity]
   interval = 2
   refine_fraction = 0.2
   coarsen_fraction = 0.3
   max_h_level = 4
  [../]

[]
```

Any suggestions would be helpful!
Thanks,
Souzan

**Comment by souzanha** [comment]
So when setting  refine_fraction = 1.0 the simulation is able to run. 
But it refines the whole mesh in the system, and does not coarsen the areas of low error. 
This of course makes everything unnecessary slow. Any ideas for why I get this type of error and how to go around it?

**Reply by AngeloCass** [reply]
Did you ever figure out a solution to this issue? I appear to be having the same problem in my simulations. When I disable adaptivity, the problem goes away. I am hoping to not have to do that, so if you figured out a way to resolve the error and keep adaptivity, please let me know! Thanks!

**Reply by permcody** [reply]
Very interesting, can either one of you push up a PR with a test case for us to take a look at?

**Reply by souzanha** [reply]
I have not been able to solve the issue sadly. It does not matter how fine the initial mesh, I always get this error. So what I do now is to disable adaptivity completely, which is quite inefficient. 

I have used adaptivity with the Allen Cahn equation before, when I use an order parameter and the adaptivity runs well with no problems. The interface mesh gets finer, and the inner phases get coarser which is exactly as expected. 

At the moment, I'm only using Cahn-Hilliard. The concentration variable is the one that changes in the system. I will try to create a simplified and stripped down version of my Cahn-Hilliard model and post it here. 

Here it is! 
https://github.com/souzanha/cahn_hilliard_test

**Comment by souzanha** [comment, marked as answer]
This seems now to be solved!

**Reply by AngeloCass** [reply]
I just saw some discussion posts saying it has been solved as well!
